### DISCUSSION TITLE
cannot create new foreign key column

### DISCUSSION THREAD

**Discussion by atiframlee** [supabase/supabase#1046] [Questions]
i already delete the previous column with foreign key, but i can still fetch data from table using deleted foreign key.
When i create new column with foreign key . the result become "More than one relationship was found for user and business"

help me out plz...

**Comment by steve-chavez** [comment]
>  the result become "More than one relationship was found for user and business"

Where do you see this error message?

While [querying foreign tables](https://supabase.io/docs/reference/javascript/select#query-foreign-tables)?

**Reply by atiframlee** [reply]
yes,while querying foreign tables.

i console.log the error message. if i remove the foreign key column, now the table has no foreign key column at all.
only then i can fetch the data. 

if i try to create the foreign key column and fetch data using the fk, the result become "More than one relationship was found for user and business".  as if the previously deleted fk column still available

![Screen Shot 2021-04-03 at 1 02 24 PM](https://user-images.githubusercontent.com/38464724/113469044-ff98ed80-947c-11eb-9929-f149965e3f70.png)

**Reply by steve-chavez** [reply]
Hm, wait. So this error happens when using the `supabase-js` library or when using the Table Editor UI?

If it's about the library, check this discussion for how to solve the error: https://github.com/supabase/supabase/discussions/809

**Comment by atiframlee** [comment]
now the issue is solved since i delete the whole table, and recreate it with exact data as before. But im worried what if the issue persist if my table contain many data(row) in future.

**Reply by steve-chavez** [reply]
Hm, it wasn't really necessary to delete the table. When having more than one foreign key between two tables, you can still use supabase-js to query/join them, but you need to add a hint as mentioned on: https://github.com/supabase/supabase/discussions/809#discussioncomment-430120
